LibSerialize.lua

```diff
@@ -376,6 +383,43 @@ local function IsFractional(value)
     return fract ~= 0
 end
 
+-- Sort compare function which is used to sort table keys to ensure that the
```

Comment:
What are your thoughts on structuring it like this instead?

```lua
local tableKeySorts = {
    ["table"] = {
        order = 1,
        cmp = function(a, b) return tostring(a) < tostring(b) end,
    },
    ["string"] = {
        order = 2,
        cmp = function(a, b) return a < b end,
    },
    ["number"] = {
        order = 3,
        cmp = function(a, b) return a < b end,
    },
    ["boolean"] = {
        order = 4,
        cmp = function(a, b) return (a and 1 or 0) < (b and 1 or 0) end,
    },
}

local function TableKeySortHelper(a, b)
    local aType, bType = type(a), type(b)
    local aSort, bSort = tableKeySorts[aType], tableKeySorts[bType]
    if not (aSort and bSort) then
        error(("Unhandled key type(s) for sorting: %s, %s"):format(aType, bType))
    end

    if aSort == bSort then
        return aSort.cmp(a, b)
    else
        return aSort.order < bSort.order
    end
end
```

Reply:
It's also not a truly stable sort for the tables-as-keys...  what if we removed support for them if you've enabled this option? In my sample code, we'd remove that from `tableKeySorts` and let the `error()` catch it.

Reply:
Thinking more about it, we don't need this at all - you can just use `table.sort`'s default sort function to sort the keys. Just add something like `self:_CanSortKey` when inserting into the map that complains (errors) about keys that aren't strings, numbers, or booleans.

Reply:
The default sort function won't handle keys of different types since it'll try to do a `<` comparison on them. I'll remove support for table keys though.

Reply:
Ah of course, we want to sort map-like tables, not array-like ones. My mistake :)